go/database/mpt/archive_trie_test.go

```diff
@@ -1690,6 +1684,42 @@ func TestArchiveTrie_FailingLiveStateUpdate_InvalidatesArchive(t *testing.T) {
 	}
 }
 
+func TestArchiveTrie_VisitTrie(t *testing.T) {
+	for _, config := range allMptConfigs {
+
+		t.Run(config.Name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			nodeVisitor := NewMockNodeVisitor(ctrl)
+			nodeVisitor.EXPECT().Visit(gomock.Any(), gomock.Any()).MinTimes(1)
```

Comment:
as I wrote previously, it should also test here that correct addresses and keys were propagated to the visitor.

also it should test that for block 0 or 1 the visitor is not called at all